micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmHeapPressureMetrics.java

```diff
@@ -125,10 +126,15 @@ private void monitor() {
 
                 if (!longLivedPoolNames.isEmpty()) {
                     final long usedAfter = longLivedPoolNames.stream()
-                        .mapToLong(pool -> after.get(pool).getUsed())
+                        .mapToLong(pool -> getUsed(after.get(pool)))
                         .sum();
-                    double maxAfter = longLivedPoolNames.stream().mapToLong(pool -> after.get(pool).getMax()).sum();
-                    lastLongLivedPoolUsageAfterGc.set(usedAfter / maxAfter);
+                    double maxAfter = longLivedPoolNames.stream().mapToLong(pool -> getMax(after.get(pool))).sum();
+                    if (usedAfter > 0 && maxAfter > 0) {
+                        lastLongLivedPoolUsageAfterGc.set(usedAfter / maxAfter);
+                    }
+                    else {
+                        lastLongLivedPoolUsageAfterGc.set(0.0);
```

Comment:
I suppose in the scenarios now covered, an exception would have been thrown before, and this would not have been updated. Maybe it's better to preserve that behavior? Although there's no way for a user to notice the issue.

Reply:
I'd err on the side of not changing this logic unless we have a good reason to (e.g. a bug report). We could wrap the get calls in `Objects.requireNonNull` to please NullAway, though it doesn't change the result (NPE thrown). What do you think?

Reply:
I think that's a good idea. Fixed it and also did a fresh rebase.